app/scripts/controllers/rif/rns/resolver/index.js

```diff
@@ -47,4 +50,56 @@ export default class RnsResolver extends RnsJsDelegate {
         }).catch(error => reject(error));
     });
   }
+  
+  /**
+   * Gets all details of a given domain name
+   * @param domainName the domain name to check (without resolver).
+   * @returns {Promise<boolean>} true if it can get all the details correctly, false otherwise.
+   */
+  getDomainDetails(domainName) {
+    const domainNameResolver = domainName + ".rsk";
+    return new Promise((resolve, reject) => {
+      this.getExpirationRemaining(domainNameResolver).then(remainingDays => { 
```

Comment:
I think it would be nice to refactor this, in order to use `async/await`, just to avoid promise `.then `cascade

Reply:
Also, i dont like the idea of invoking contracts from the controller. I find the contorller as a top level component. For that we have the service layer. 

Lets discuss that too

Reply:
If he do that it can be a little problematic, since if you use await you will need to put your function as async always. 
I think this is a matter of taste, it depends on the situation, maybe you run a method that needs to do something on the callback but you don't want to wait for it and with async/await you can't do that because you are waiting for the response.
I know that in this case is a lot so maybe we should change all the delegate expose api methods to async instead, and then start thinking about when is good to use async/await and when is not, probably we will need to keep a balance on that.

Reply:
I dont have in mind any situation where async/await must not be used, but sure we can discuss that.

Take into account that when the callback executes is after the principal operation finish, so is the same as wait for the result with `await`

Reply:
Why you should force to use async to the caller always? that's what i don't understand, i mean if you use async you are forcing your caller to use await if he wants the result, instead with promises you can await or just manage the response with then, is matter of taste, if not, promises shouldn't exists and no one should use them, i think is just different schools that's all.

Reply:
Hey guys, the code has change a little, now it's using promises.all for this.